app/controllers/api_controller/providers.rb

```diff
@@ -5,9 +5,14 @@ module Providers
     CREDENTIALS_ATTR  = "credentials"
     AUTH_TYPE_ATTR    = "auth_type"
     DEFAULT_AUTH_TYPE = "default"
-    ENDPOINT_ATTRS    = %w(hostname ipaddress port)
+    ENDPOINT_ATTRS    = %w(hostname ipaddress port connection_configurations_array).freeze
```

Comment:
an answer for 
`GET /api/providers/:id?attributes=connection_configurations,connection_configurations_array`

```
{
  "href": "http://localhost:3000/api/providers/139",
  "id": 139,
  "name": "LABz",
  "created_on": "2016-07-05T12:43:52.458Z",
  "updated_on": "2016-07-05T12:43:52.458Z",
  "guid": "20dd91ec-42ae-11e6-b9b4-507b9d788844",
  "zone_id": 1,
  "type": "ManageIQ::Providers::Openshift::ContainerManager",
  "api_version": "v1",
  "tenant_id": 1,
  "connection_configurations": {
    "table": {
      "default": {
        "table": {
          "endpoint": {
            "id": 334,
            "role": "default",
            "ipaddress": null,
            "hostname": "yzamir-centos7-1.eng.lab.tlv.redhat.comxz",
            "port": 8443,
            "resource_type": "ExtManagementSystem",
            "resource_id": 139,
            "created_at": "2016-07-05T12:43:52.461Z",
            "updated_at": "2016-07-05T12:43:52.461Z",
            "verify_ssl": 1,
            "url": null,
            "security_protocol": null,
            "api_version": null,
            "path": null
          },
          "authentication": null
        }
      },
      "hawkular": {
        "table": {
          "endpoint": {
            "id": 335,
            "role": "hawkular",
            "ipaddress": null,
            "hostname": "yzamir-centos7-1.eng.lab.tlv.redhat.comxz",
            "port": 443,
            "resource_type": "ExtManagementSystem",
            "resource_id": 139,
            "created_at": "2016-07-05T12:43:52.463Z",
            "updated_at": "2016-07-05T12:43:52.463Z",
            "verify_ssl": 1,
            "url": null,
            "security_protocol": null,
            "api_version": null,
            "path": null
          },
          "authentication": {
            "id": 236,
            "name": "ManageIQ::Providers::Openshift::ContainerManager LABz",
            "authtype": "hawkular",
            "userid": "_",
            "password": null,
            "resource_id": 139,
            "resource_type": "ExtManagementSystem",
            "created_on": "2016-07-05T12:43:52.611Z",
            "updated_on": "2016-07-05T12:43:53.253Z",
            "last_valid_on": null,
            "last_invalid_on": "2016-07-05T12:43:53.252Z",
            "credentials_changed_on": "2016-07-05T12:43:52.611Z",
            "status": "Unreachable",
            "status_details": "getaddrinfo: No address associated with hostname",
            "auth_key": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJtYW5hZ2VtZW50LWluZnJhIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6Im1hbmFnZW1lbnQtYWRtaW4tdG9rZW4tcnAxZ3giLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoibWFuYWdlbWVudC1hZG1pbiIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMzODE3NWU5LTM5NGItMTFlNi1iOWFkLTAwMWE0YTIzMTRkNSIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDptYW5hZ2VtZW50LWluZnJhOm1hbmFnZW1lbnQtYWRtaW4ifQ.nORVyramYixUz5_Fg5j-A2FILYnUmPzdHiXWRXfQpdAfaMe-YoKjsZOhCDvaOyoNxjIQ3MTIUU48ZZ8Wk9j-GJUkJhWAU48rFhPi-yw8fv8N11jlx9p2p6wfseLVQt11rmVyAUoTVQLGeo-RIWPyegR1DNVIDW2ciPGeKBiuJKHxDJBS3whuQ2apf-6J75_EfEnCBkhhwKoxvtlGhfPdrhJwy16kmw9p6J0cRvhjr31wZo-NY-YI9SF4JVbQD2Tg08ohSZTjQiehTlv0Fm8MJ4x_4RHmhVnYodUOo6HQoDHjSscJMnwBDC2R-QHFET145aVLhLNVZDpiFivUHcgp3A",
            "fingerprint": null,
            "service_account": null,
            "challenge": null,
            "login": null,
            "public_key": null,
            "htpassd_users": [],
            "ldap_id": [],
            "ldap_email": [],
            "ldap_name": [],
            "ldap_preferred_user_name": [],
            "ldap_bind_dn": null,
            "ldap_insecure": null,
            "ldap_url": null,
            "request_header_challenge_url": null,
            "request_header_login_url": null,
            "request_header_headers": [],
            "request_header_preferred_username_headers": [],
            "request_header_name_headers": [],
            "request_header_email_headers": [],
            "open_id_sub_claim": null,
            "open_id_user_info": null,
            "open_id_authorization_endpoint": null,
            "open_id_token_endpoint": null,
            "open_id_extra_scopes": [],
            "open_id_extra_authorize_parameters": null,
            "certificate_authority": null,
            "google_hosted_domain": null,
            "github_organizations": [],
            "rhsm_sku": null,
            "rhsm_pool_id": null,
            "rhsm_server": null
          }
        }
      },
      "roles": [
        "default",
        "hawkular"
      ]
    },
    "modifiable": true
  },
  "connection_configurations_array": [
    {
      "endpoint": {
        "id": 334,
        "role": "default",
        "hostname": "yzamir-centos7-1.eng.lab.tlv.redhat.comxz",
        "port": 8443,
        "resource_type": "ExtManagementSystem",
        "resource_id": 139,
        "created_at": "2016-07-05T12:43:52.461Z",
        "updated_at": "2016-07-05T12:43:52.461Z",
        "verify_ssl": 1
      },
      "authentication": {
        "id": 235,
        "name": "ManageIQ::Providers::Openshift::ContainerManager LABz",
        "authtype": "bearer",
        "userid": "_",
        "resource_id": 139,
        "resource_type": "ExtManagementSystem",
        "created_on": "2016-07-05T12:43:52.465Z",
        "updated_on": "2016-07-05T12:43:53.207Z",
        "last_invalid_on": "2016-07-05T12:43:53.206Z",
        "credentials_changed_on": "2016-07-05T12:43:52.465Z",
        "status": "Unreachable",
        "status_details": "getaddrinfo: No address associated with hostname",
        "auth_key": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJtYW5hZ2VtZW50LWluZnJhIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6Im1hbmFnZW1lbnQtYWRtaW4tdG9rZW4tcnAxZ3giLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoibWFuYWdlbWVudC1hZG1pbiIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMzODE3NWU5LTM5NGItMTFlNi1iOWFkLTAwMWE0YTIzMTRkNSIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDptYW5hZ2VtZW50LWluZnJhOm1hbmFnZW1lbnQtYWRtaW4ifQ.nORVyramYixUz5_Fg5j-A2FILYnUmPzdHiXWRXfQpdAfaMe-YoKjsZOhCDvaOyoNxjIQ3MTIUU48ZZ8Wk9j-GJUkJhWAU48rFhPi-yw8fv8N11jlx9p2p6wfseLVQt11rmVyAUoTVQLGeo-RIWPyegR1DNVIDW2ciPGeKBiuJKHxDJBS3whuQ2apf-6J75_EfEnCBkhhwKoxvtlGhfPdrhJwy16kmw9p6J0cRvhjr31wZo-NY-YI9SF4JVbQD2Tg08ohSZTjQiehTlv0Fm8MJ4x_4RHmhVnYodUOo6HQoDHjSscJMnwBDC2R-QHFET145aVLhLNVZDpiFivUHcgp3A",
        "htpassd_users": [],
        "ldap_id": [],
        "ldap_email": [],
        "ldap_name": [],
        "ldap_preferred_user_name": [],
        "request_header_headers": [],
        "request_header_preferred_username_headers": [],
        "request_header_name_headers": [],
        "request_header_email_headers": [],
        "open_id_extra_scopes": [],
        "github_organizations": [],
        "role": "bearer"
      }
    },
    {
      "endpoint": {
        "id": 335,
        "role": "hawkular",
        "hostname": "yzamir-centos7-1.eng.lab.tlv.redhat.comxz",
        "port": 443,
        "resource_type": "ExtManagementSystem",
        "resource_id": 139,
        "created_at": "2016-07-05T12:43:52.463Z",
        "updated_at": "2016-07-05T12:43:52.463Z",
        "verify_ssl": 1
      },
      "authentication": {
        "id": 236,
        "name": "ManageIQ::Providers::Openshift::ContainerManager LABz",
        "authtype": "hawkular",
        "userid": "_",
        "resource_id": 139,
        "resource_type": "ExtManagementSystem",
        "created_on": "2016-07-05T12:43:52.611Z",
        "updated_on": "2016-07-05T12:43:53.253Z",
        "last_invalid_on": "2016-07-05T12:43:53.252Z",
        "credentials_changed_on": "2016-07-05T12:43:52.611Z",
        "status": "Unreachable",
        "status_details": "getaddrinfo: No address associated with hostname",
        "auth_key": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJtYW5hZ2VtZW50LWluZnJhIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6Im1hbmFnZW1lbnQtYWRtaW4tdG9rZW4tcnAxZ3giLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoibWFuYWdlbWVudC1hZG1pbiIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImMzODE3NWU5LTM5NGItMTFlNi1iOWFkLTAwMWE0YTIzMTRkNSIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDptYW5hZ2VtZW50LWluZnJhOm1hbmFnZW1lbnQtYWRtaW4ifQ.nORVyramYixUz5_Fg5j-A2FILYnUmPzdHiXWRXfQpdAfaMe-YoKjsZOhCDvaOyoNxjIQ3MTIUU48ZZ8Wk9j-GJUkJhWAU48rFhPi-yw8fv8N11jlx9p2p6wfseLVQt11rmVyAUoTVQLGeo-RIWPyegR1DNVIDW2ciPGeKBiuJKHxDJBS3whuQ2apf-6J75_EfEnCBkhhwKoxvtlGhfPdrhJwy16kmw9p6J0cRvhjr31wZo-NY-YI9SF4JVbQD2Tg08ohSZTjQiehTlv0Fm8MJ4x_4RHmhVnYodUOo6HQoDHjSscJMnwBDC2R-QHFET145aVLhLNVZDpiFivUHcgp3A",
        "htpassd_users": [],
        "ldap_id": [],
        "ldap_email": [],
        "ldap_name": [],
        "ldap_preferred_user_name": [],
        "request_header_headers": [],
        "request_header_preferred_username_headers": [],
        "request_header_name_headers": [],
        "request_header_email_headers": [],
        "open_id_extra_scopes": [],
        "github_organizations": [],
        "role": "hawkular"
      }
    }
  ],
  "actions": [
    {
      "name": "edit",
      "method": "post",
      "href": "http://localhost:3000/api/providers/139"
    },
    {
      "name": "refresh",
      "method": "post",
      "href": "http://localhost:3000/api/providers/139"
    },
    {
      "name": "delete",
      "method": "post",
      "href": "http://localhost:3000/api/providers/139"
    },
    {
      "name": "delete",
      "method": "delete",
      "href": "http://localhost:3000/api/providers/139"
    }
  ]
}
```